presto-main/src/main/java/com/facebook/presto/failureDetector/HeartbeatFailureDetector.java

```diff
@@ -143,6 +148,31 @@ public void shutdown()
                 .collect(toImmutableSet());
     }
 
+    @Override
+    public State getState(HostAddress hostAddress)
+    {
+        for (MonitoringTask task : tasks.values()) {
+            if (hostAddress.equals(fromUri(task.uri))) {
+                if (!task.isFailed()) {
+                    return State.ALIVE;
```

Comment:
The reason for the ordering is because `ALIVE` and `UNKNOWN` are the two states will always exits. And we can add more states as we can identify more reasons :)